ethers-solc/src/config.rs

```diff
@@ -146,6 +146,69 @@ impl ProjectPathsConfig {
     pub fn find_libs(root: impl AsRef<Path>) -> Vec<PathBuf> {
         vec![utils::find_fave_or_alt_path(root, "lib", "node_modules")]
     }
+
+    /// Flatten all file imports into a single string
+    pub fn flatten(&self, target: &PathBuf) -> Result<String> {
+        tracing::trace!("flattening file");
+        let graph = Graph::resolve(&self)?;
+
+        struct Flattener<'a> {
+            f: &'a dyn Fn(&Flattener, &PathBuf) -> Result<String>,
+        }
+        let flatten = Flattener {
+            f: &|flattener, target| {
+                let target_dir = target.parent().ok_or(SolcError::msg(format!(
+                    "failed to get parent directory for \"{:?}\"",
+                    target.display()
+                )))?;
+                let target_index = graph.files().get(target).ok_or(SolcError::msg(format!(
+                    "cannot resolve file at \"{:?}\"",
+                    target.display()
+                )))?;
+                let target_node = graph.node(*target_index);
+
+                let mut imports = target_node.imports().clone();
+                imports.sort_by(|a, b| b.loc().0.cmp(&a.loc().0));
+
+                let content = target_node.content().bytes().collect::<Vec<_>>();
+                let mut extended = vec![];
+                let mut curr_import = imports.pop();
+
+                let mut i = 0;
+                while i < content.len() {
+                    if let Some(ref import) = curr_import {
+                        let (start, end) = import.loc();
+                        if i == start {
+                            let import_path =
+                                utils::resolve_import_component(import.path(), target_dir, &self)?;
+                            let import_content = (flattener.f)(flattener, &import_path)?;
+                            let import_content =
+                                utils::RE_SOL_PRAGMA_VERSION.replace_all(&import_content, "");
```

Comment:
@mattsse the location for the pragma version directive is also not exposed, do we want to add another PR to solang?

Reply:
yeah sgtm!